wee_slack.py

```diff
@@ -2644,6 +2647,39 @@ def process_message(message_json, eventrouter, store=True, **kwargs):
             channel.store_message(message, team)
         dbg("NORMAL REPLY {}".format(message_json))
 
+    if download:
+        download_files(message_json, **kwargs)
+
+
+def download_files(message_json, **kwargs):
+    team = kwargs["team"]
+    download_location = config.get_string('files_download_location')
+    if not download_location:
+        return
+
+    def fileout_iter(path):
+        yield path
+        main, ext = os.path.splitext(path)
+        for i in count(start=1, step=1):
+            yield main + "-{}".format(i) + ext
+
+    for f in message_json.get('files', []):
+        if f.get('mode') == 'tombstone':
+            continue
+
+        filename = '{}_{}.{}'.format(team.preferred_name, f['title'], f['filetype'])
```

Comment:
I still think we shouldn't append the file type if it's already there, so we don't end up with `.png.png`. E.g.:

```
        filetype = '' if f['title'].endswith(f['filetype']) else '.' + f['filetype']
```